File: graphql-java-servlet/src/main/java/com/axxes/graphql/java/servlet/dto/EmployeeDto.java
package com.axxes.graphql.java.servlet.dto;

import java.util.Set;

public class EmployeeDto {
    private final Long id;
    private final String firstName;
    private final String lastName;
    private final String email;
    private final Set<CompetenceCenterDto> competenceCenters;

    public EmployeeDto(Long id, String firstName, String lastName, String email, Set<CompetenceCenterDto> competenceCenters) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.competenceCenters = competenceCenters;
    }

    public Long getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getEmail() {
        return email;
    }

    public Set<CompetenceCenterDto> getCompetenceCenters() {
        return competenceCenters;
    }
}

File: graphql-kotlin-spring-server/src/main/kotlin/com/axxes/graphql/kotlin/spring/server/dto/EmployeeDto.kt
package com.axxes.graphql.kotlin.spring.server.dto

data class EmployeeDto(
        val id: Long,
        val firstName: String,
        val lastName: String,
        val email: String,
        val competenceCenters: List<CompetenceCenterDto>
)

enum class CompetenceCenterDto {
    JAVA, DOTNET, FRONTEND, TESTING, INFRA
}